.github/workflows/ci.yml

```
@@ -0,0 +1,32 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: ci
+
+on: [push, pull_request]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

Unless you just want it to fine hard syntactical errors?
```suggestion
        flake8 . --count --max-complexity=10 --max-line-length=127 --statistics
```

---

I just adopted this with only minimal modification from: https://github.com/actions/starter-workflows/blob/main/ci/python-package.yml

I suggest that we fix style issues in a separate PR.

---

Yes, I know that that's how they suggest to do it, but my point here is that when all errors are considered as warnings this step will _never_ fail, I think. So it's basically useless. Unless you go and check the output every time.

---

Ok, but if we make it fail, then we also have to fix the style issues as part of this PR. Is that what you would prefer?

---

Ah, I see. Well ... since you're adding it now you might as well. Although I would definitely keep that as a separate commit. I think that's up to you. Perhaps try to run it and see how many changes there are, and then take a decision based on that? I don't mind adding it to this PR, since the workflow is added in this PR as well.

---

I've just removed the whole GA workflow and thus linting. No need to create unnecessary work here. This is supposed to be a lightweight change.